src/test/testLogger.ts

```diff
@@ -40,12 +44,17 @@ export class TestLogger implements Logger {
             .map(loggedEntry => loggedEntry.entry)
     }
 
+    public writeLoggedEntriesToFile(entry: Loggable) {
+        appendFileSync(testLogOutput, `${entry}\n`, 'utf8')
+    }
+
     private addLoggedEntries(logLevel: LogLevel, entries: Loggable[]) {
         entries.forEach(entry => {
             this.loggedEntries.push({
                 logLevel,
                 entry
             })
+            this.writeLoggedEntriesToFile(entry)
```

Comment:
why not build a multiline string instead of opening/closing the file many times? 

It's not only a micro-optimization: it is at least as readable, and also avoids potential races if another process is writing to the same file (causing interleaved lines), and reduces the opportunity for read/write failures.

Reply:
I mean sure we can just flush it on test end, changed